File: 9.rb
def triplet
  (1..1000).each do |a|
    (a..999).each do |b|
      c = 1000 - b - a
      return [a,b,c] if a**2 + b**2 == c**2
    end
  end
end

puts triplet
puts triplet.inject(&:+)
puts triplet[0]**2 + triplet[1]**2
puts triplet[2]**2

puts triplet.inject(&:*)

File: 09/9.exs
sums = Enum.flat_map(1..1000, fn
  a ->
    Enum.map(a..999, fn
      b ->
        {a, b, 1000 - b - a}
    end)
end)

triplet = sums
          |> Enum.filter(fn {a, b, c} -> a*a + b*b == c*c end)
          |> Enum.map(fn {a, b, c} -> a * b * c end)

IO.puts (inspect triplet)
